packages/expo-file-system/ios/Next/FileSystemNextModule.swift

```diff
@@ -6,6 +6,40 @@ public final class FileSystemNextModule: Module {
   public func definition() -> ModuleDefinition {
     Name("FileSystemNext")
 
+    AsyncFunction("download") { (url: URL, to: FileSystemPath, promise: Promise) in
+      let downloadTask = URLSession.shared.downloadTask(with: url) { urlOrNil, responseOrNil, errorOrNil in
+        guard errorOrNil == nil else {
+          return promise.reject(UnableToDownloadException(errorOrNil?.localizedDescription ?? "unspecified error"))
+        }
+        guard let httpResponse = responseOrNil as? HTTPURLResponse else {
+          return promise.reject(UnableToDownloadException("no response"))
+        }
+        guard httpResponse.statusCode >= 200 && httpResponse.statusCode < 300 else {
```

Comment:
Just a small suggestion to make it more "Swifty"
```suggestion
        guard (200..<300).contains(httpResponse.statusCode) else {
```